### ISSUE TITLE
Get oxipng into linux distributions and Homebrew

### ISSUE THREAD

**Issue by shssoichiro** [shssoichiro/oxipng#69]
As `cargo install` is not recommended to be used for installing end-user binaries, we should move to get oxipng listed in Linux distributions and Homebrew.

We may not have immediate control over some of these.

- [ ] Homebrew
- [ ] Ubuntu
- [ ] Debian
- [ ] Gentoo
- [ ] Arch
- [ ] Centos

**Comment by CounterPillow** [comment]
Good luck, rust's static compilation means no distro will want to touch this with a 10 foot pole

**Comment by shssoichiro** [comment]
Ripgrep is at least in Homebrew and Gentoo. For Arch there's the AUR. The other distros will probably be a harder sell.

**Comment by corppneq** [comment]
You could use AppImage: https://github.com/AppImage/AppImageKit/

**Comment by leo-unglaub** [comment]
I have build a package for `Alpine Linux` and `OpenBSD`. I will submit them shortly to the official repositories.

**Comment by magicgoose** [comment]
> Good luck, rust's static compilation means no distro will want to touch this with a 10 foot pole

pngquant (later versions) is written in rust as well, and it's present in many distros: https://pkgs.org/download/pngquant

**Comment by CounterPillow** [comment]
@magicgoose that pngquant must not be the one you're referring to, because the one you linked is written in C99.

**Comment by magicgoose** [comment]
@CounterPillow I'm referring to https://github.com/kornelski/pngquant
It is partially written in Rust. (not completely — my mistake)

**Comment by Calinou** [comment]
Just a heads-up for Windows users: oxipng is now available in [Scoop](https://scoop.sh/) ([commit](https://github.com/lukesampson/scoop/commit/664e135d3f1f11e104ae0b53135b0ff98277b5d0)) :smiley:

**Comment by aaronfranke** [comment]
What's the status of this? What is the recommended way to install oxipng?

**Comment by adiabatic** [comment]
It's been in Homebrew for a while.

**Comment by CounterPillow** [comment]
It's also in the official Arch Linux repositories https://archlinux.org/packages/community/x86_64/oxipng/

Since Debian and Ubuntu ship other rust utilities now (e.g. the excellent ripgrep) I imagine it's only a matter of time before a Debian or Ubuntu contributor takes interest and packages it.

**Comment by remilauzier** [comment]
It's now in fedora. https://src.fedoraproject.org/rpms/rust-oxipng

**Comment by jn64** [comment]
It was orphaned in Fedora since 2022-04 :(

**Comment by GreenReaper** [comment]
I agree it'd be great to see this in Debian, which would perhaps help get into Ubuntu as well. [There's been a request open since 2018](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=913285). However, it'll probably require someone active in the project to get involved as they have security maintenance requirements.

[The Arch Linux package is now over here](https://archlinux.org/packages/extra/x86_64/oxipng/). Technically, if you tar xvf the package and install it to /usr/bin, it _seems_ to work on Debian Bookworm. (It may rely on certain libraries.)

**Comment by AlexTMjugador** [comment]
> Technically, if you tar xvf the package and install it to /usr/bin, it seems to work on Debian Bookworm. (It may rely on certain libraries.)

OxiPNG distributes fully statically linked Linux binaries, whose release archive names end in `-musl`. These binaries have no dependencies on any distro's `libc`, so if you want OxiPNG binaries that are portable to almost any Linux environment _right now_ without any hassle (including Debian, and more exotic things like Alpine Linux and Android), that's your best bet.

> [...] However, it'll probably require someone active in the project to get involved as they have security maintenance requirements.

I would love to see OxiPNG on Debian as well, but frankly, the amount of work required to play by the Debian repository rules is significant, and I am not convinced that it is worth the effort for a core OxiPNG developer to also become a Debian maintainer, given the apparent lack of interest so far from any Debian maintainer to even consider getting in touch with this project.

If anyone with more experience with Debian packaging has a concrete plan to make it work well ("well" as in that the package has `man` pages and proper data and metadata, is available for different architectures, has backports of newer OxiPNG versions to older distro releases, etc.) without much effort on our part, please feel free to chime in!

**Comment by andrews05** [comment]
@calumapplepie Thanks for the info! It's been on the todo list to get a man page, I can at least look at getting that sorted...

**Comment by musicinmybrain** [comment]
> update: it was unretired in 2024-02. Packages are again available in Fedora 38 and up, as well as EPEL 9.

I’m the new maintainer. Feel free to ping me any time you need to talk about how oxipng is packaged in Fedora and/or EPEL.

**Comment by Chubercik** [comment]
> What's the status of this? Will there be Ubuntu packages anytime soon? Perhaps a PPA?

Bump; any news in this department?

**Comment by andrews05** [comment]
@Chubercik We have a man page now. It also looks like oxipng is available on Gentoo, but no movement on Ubuntu/Debian afaik.

**Comment by AlexTMjugador** [comment]
I think it's useful to point out that since this issue was created, an infographic powered by `repology.org` has been added to the repository README to show which Oxipng version is being packaged in which repositories, so anyone can use this to track packaging status in real time.

**Comment by TPS** [comment]
What's _very_ nice on the [linked repology page](https://repology.org/project/oxipng/versions) is that platform maintainers are linked.  Ubuntu/Debian is conspicuously entirely absent from that page.  Interested, @Chubercik @calumapplepie?

**Comment by Chubercik** [comment]
> What's _very_ nice on the [linked repology page](https://repology.org/project/oxipng/versions) is that platform maintainers are linked. Ubuntu/Debian is conspicuously entirely absent from that page. Interested, @Chubercik @calumapplepie?

I don't really have nearly enough knowledge about the Debian/Ubuntu ecosystem to tackle this myself; for now, I've followed the "Requesting a new package for Ubuntu" section of https://wiki.ubuntu.com/UbuntuDevelopment/NewPackages and filed a [needs-packaging] bug report here (that also mentions the Debian WNPP):

- https://bugs.launchpad.net/ubuntu/+bug/2072673

**Comment by Chubercik** [comment]
So I fiddled with packaging `oxipng` for Debian today..

![image](https://github.com/user-attachments/assets/15a8f8b8-e886-4f49-832c-bce66fe07c91)
(screenshot courtesy of [cargo-debstatus](https://github.com/kpcyrd/cargo-debstatus))

Unfortunately, a few dependencies are outdated (or absent) and thus require to be taken care of first.

**Comment by TPS** [comment]
`OxiPNG` has been [added to the TUR](https://github.com/termux-user-repository/tur/pull/2134) yesterday for convenience & semi-official support! 🎉

**Comment by Chubercik** [comment]
Debian update:

<img width="749" height="348" alt="Image" src="https://github.com/user-attachments/assets/494163f6-5972-4eaf-9e1c-8ac83f5064a7" />
